src/atoms/Button/Button.native.js

```diff
@@ -1,35 +1,279 @@
-import React from 'react';
+import React, { useContext } from 'react';
 import PropTypes from 'prop-types';
-import styled from 'styled-components/native';
-import { TouchableOpacity, Text } from 'react-native';
+import styled, { ThemeContext } from 'styled-components/native';
+import { TouchableHighlight } from 'react-native';
 
-const StyledButton = styled(TouchableOpacity)(
-  (props) => `
-  background-color: ${props.theme.colors.shade[900]};
-  padding: 14px;
-  border-radius: 6px;
-  margin-top: 7px;
-`,
-);
+import Icon from '../Icon';
+import Space from '../Space';
+import Text from '../Text';
+import View from '../View';
+import Flex from '../Flex';
+import automation from '../../_helpers/automation-attributes';
+import { getColor, makePxValue, getVariantColorKeys } from '../../_helpers/theme';
+
+const styles = {
+  fontColor({ variant, variantColor, disabled }) {
+    switch (variant) {
+      case 'primary':
+      default:
+        if (disabled) {
+          return 'white.950'; //TODO: mapping not present(EX/white 500) for figma in sheet.
+        }
+        return 'white.900';
+      case 'secondary':
+        if (disabled) {
+          return 'shade.940';
+        }
+        return `${variantColor || 'primary'}.800`;
+      case 'tertiary':
+        if (disabled) {
+          return 'shade.940';
+        }
+        return `${variantColor || 'primary'}.800`;
+    }
+  },
+  backgroundColor({ variant, variantColor, disabled, theme }) {
+    switch (variant) {
+      case 'primary':
```

Comment:
Can we not have short circuits and use everything explicitly for better readability? I remember @deorukhkarchaitanya doing the same in one of the components. It surely saves you few micro seconds but readabilty and debugging experience goes for a toss